crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1512,7 +1512,23 @@ impl Room {
 
                 if !self.are_members_synced() {
                     self.sync_members().await?;
-                    // TODO query keys here?
+
+                    // Query keys in case we don't have them for newly synced members.
+                    let olm = self.client.olm_machine().await;
+                    let olm = olm.as_ref().expect("Olm machine wasn't started");
+
+                    let members = self
+                        .client
+                        .store()
+                        .get_user_ids(self.room_id(), RoomMemberships::ACTIVE)
+                        .await?;
+
+                    let (req_id, request) =
+                        olm.query_keys_for_users(members.iter().map(|owned| owned.borrow()));
+
+                    if let Err(err) = self.client.keys_query(&req_id, request.device_keys).await {
```

Comment:
I think we should likely abort here? We're almost certain to produce a UTD othewrise.